### PR TITLE
Upgrade to RxJava 3

### PR DESCRIPTION
**Pull request by BraisGabin**
All the imports were changed using [this script](https://github.com/BraisGabin/RxJava3-migration-tool).

This is a breaking change for sure but I didn't change any package or coordinates. Should I change those too?

Related with #531

### CODE REVIEW

**Comment by JakeWharton** [comment]
Yes we will need to change the package name and maven groupId to use a "4" instead of a "3".

**Comment by JakeWharton** [comment]
Please rebase on master with RxJava 3 final and we can get this merged.

**Comment by BraisGabin** [comment]
Rebased

**Comment by Thomvis** [comment]
Is there a way I can help with this PR? I'd love to get a RxJava 3 compatible release of RxBinding.

**Comment by BraisGabin** [comment]
I think that this PR is ready to be merged. If I need to do something else I'd be happy to do it.

**Comment by JakeWharton** [comment]
I will spend time upgrading all my Rx libraries to 3 soon.

File: rxbinding/build.gradle [hunk 1]
```diff
@@ -26,6 +26,9 @@ android {
     sourceCompatibility JavaVersion.VERSION_1_8
     targetCompatibility JavaVersion.VERSION_1_8
   }
+  kotlinOptions {
+    jvmTarget = '1.8'
+  }
```
[the remark this answers is not in the document]
Comment (ZacSweers) [reply rxbinding/build.gradle:31]: For sure do this in the others. You could probably add something in the root project to share this configuration

```gradle
subprojects {
  tasks.withType(KotlinCompile).configureEach {
    kotlinOptions.jvmTarget = "1.8"
  }
}
```

File: build.gradle [hunk 2]
```diff
@@ -28,8 +28,8 @@ buildscript {
 
     'googleMaterial': 'com.google.android.material:material:1.0.0',
 
-    'rxJava': 'io.reactivex.rxjava2:rxjava:2.2.10',
-    'rxAndroid': 'io.reactivex.rxjava2:rxandroid:2.1.1',
+    'rxJava': 'io.reactivex.rxjava3:rxjava:3.0.0',
```
Comment (ZacSweers) [inline build.gradle:31]: Might as well pick up the latest patch

```suggestion
    'rxJava': 'io.reactivex.rxjava3:rxjava:3.0.2',
```

**Pull request merged**